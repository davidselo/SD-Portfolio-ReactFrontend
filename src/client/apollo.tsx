export default class apolloClient {
    useQuery = () => {
        return 'use Query from Apollo';
    };
}
