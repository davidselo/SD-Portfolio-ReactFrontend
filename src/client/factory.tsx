import apolloClient from './apollo';
export const getClient = () => {
    return new apolloClient();
};

export default getClient;
