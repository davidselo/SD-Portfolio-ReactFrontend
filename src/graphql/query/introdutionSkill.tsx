import { gql } from '@apollo/client';

export const GET_SKILLS = gql`
  query{
    introductionSkills{
        description
    }
}
`;
