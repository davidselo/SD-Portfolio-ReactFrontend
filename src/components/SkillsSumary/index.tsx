import React from 'react';
import {GET_SKILLS} from '../../graphql/query/introdutionSkill';
import {useQuery} from '@apollo/client';
import {Skills} from '../../contracts/Skills';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Stack from '@mui/material/Stack';

const SkillsSumary: React.FC = () => {
    const {loading, error, data} = useQuery<Skills>(GET_SKILLS);
    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error! ${error.message}</p>;
    return (
        <>
            <Grid
                container
                spacing={0}
                alignContent="center"
                rowSpacing={5}
                sx={{
                    backgroundColor: '#98B9F2',
                    padding: '30px 30px 30px 30px',
                    minHeight: '50vh',
                }}>
                <Grid item xl={7} lg={8} md={6} sm={12} xs={12}>
                    <Stack spacing={3}>
                        <Typography variant="h4">
                            Why Do I enjoy what I do?
                        </Typography>
                        {data.introductionSkills.map((skill, index) => (
                            <Typography variant="h5" key={index}>
                                {skill.description}
                            </Typography>
                        ))}
                    </Stack>
                </Grid>
                <Grid item xl={5} lg={4} md={6} sm={12} xs={12}>
                    <Box textAlign="center" paddingTop="50">
                        <Box
                            sx={{
                                display: 'inline',
                                color: '#e27e02',
                            }}>
                            <Typography
                                component="span"
                                sx={{
                                    fontSize: '11rem',
                                    textAlign: 'center',
                                    display: 'inline',
                                    color: '#e27e02',
                                    lineHeight: '0.9',
                                }}>
                                12
                            </Typography>
                        </Box>
                        <Box sx={{display: 'inline', color: '#e27e02'}}>
                            <Typography component="span">years</Typography>
                        </Box>
                        <Box sx={{fontSize: '1.4rem', textAlign: 'center'}}>
                            <Typography component="span">
                                of exciting web development journey
                            </Typography>
                        </Box>
                    </Box>
                </Grid>
            </Grid>
        </>
    );
};

export default SkillsSumary;
