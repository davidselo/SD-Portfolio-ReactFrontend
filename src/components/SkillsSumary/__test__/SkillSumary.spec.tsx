import React from 'react';
import '@testing-library/jest-dom';
import {render, screen} from '@testing-library/react';
import SkillsSumary from '../index';
import {MockedProvider} from '@apollo/client/testing';
import {GET_SKILLS} from '../../../graphql/query/introdutionSkill';
import {GraphQLError} from 'graphql';

describe('Skills Sumary Component', () => {
    test('Component renders the block with Mock data', async () => {
        // Arrange
        const mocks = [
            {
                request: {
                    query: GET_SKILLS,
                },
                result: {
                    data: {
                        introductionSkills: [
                            {description: 'Description 1'},
                            {description: 'Description 2'},
                            {description: 'Description 3'},
                            {description: 'Description 4'},
                            {description: 'Description 5'},
                        ],
                    },
                },
            },
        ];
        // Act
        const {container} = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <SkillsSumary />
            </MockedProvider>,
        );
        // Assert
        expect(await screen.findByText('Loading...')).toBeInTheDocument();
        expect(await screen.findByText('Description 1')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    test('Component when got error on GraphQL', async () => {
        // Arrange
        const mocks = [
            {
                request: {
                    query: GET_SKILLS,
                },
                result: {
                    errors: [new GraphQLError('code 1')],
                },
            },
        ];
        const {container} = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <SkillsSumary />
            </MockedProvider>,
        );
        expect(await screen.findByText('Error! code 1')).toBeInTheDocument();
    });
});
