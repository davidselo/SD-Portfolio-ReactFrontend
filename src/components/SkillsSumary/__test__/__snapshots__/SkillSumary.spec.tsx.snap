// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills Sumary Component Component renders the block with Mock data 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-1chm1sj-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-8 MuiGrid-grid-xl-7 css-1p3rggy-MuiGrid-root"
    >
      <div
        class="css-1lwbda4-MuiStack-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          Why Do I enjoy what I do?
        </h4>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Description 1
        </h5>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Description 2
        </h5>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Description 3
        </h5>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Description 4
        </h5>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Description 5
        </h5>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-5 css-frozg5-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1pmgad6"
      >
        <div
          class="MuiBox-root css-1tysxka"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-w71u00-MuiTypography-root"
          >
            12
          </span>
        </div>
        <div
          class="MuiBox-root css-1tysxka"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            years
          </span>
        </div>
        <div
          class="MuiBox-root css-ulowpu"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            of exciting web development journey
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
